bloom-housing/bloom
# feat: new what to expect defaults

# Pull Request Template

## Issue Overview

This PR addresses #3031


## Description

This PR adds new default text in the what to expect field on a listing based on the type of the listing (either (1) first come first served (2) lottery or (3) waitlist). At the moment, the listing form allows for a waitlist (determined by the radio field on units) to exist on both a lottery or a first come first serve listing. The listing type [determination](https://exygy.slack.com/archives/C01Q4QG5R8Q/p1662650619602509?thread_ts=1662498643.729889&cid=C01Q4QG5R8Q) is that lottery overrides waitlist but waitlist overrides fcfs.

We do want to keep any what to expect fields that are custom. To do this, every field that is still the default is set to null. The frontend checks for an existing value and otherwise sets the new strings on the frontend.

## How Can This Be Tested/Reviewed?

Reseed on dev. Add a listing with a what to expect value of the default. Run the migration on this branch. Ensure the what to expect value was set to null and that other custom ones were kept. Verify that changing the listing type does change the strings.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
